pykumo/py_kumo.py

```diff
@@ -354,6 +355,13 @@ def get_current_sensor_temperature(self):
             val = None
         return val
 
+    def get_unit_schedule(self):
+        """ Retrieve the program (UnitSchedule) from sensor, if any. """
+        unit_schedule = UnitSchedule(self)
+        unit_schedule.fetch()
```

Comment:
I wonder if  each PyKumo object should own its own UnitSchedule object and call `fetch` from someplace in `update_status`. Then this method can be non-blocking like the other getters.

There could be an optional `use_schedule` option to the PyKumo constructor (defaulting to False, at least for the time being) that controls whether it does this.

Reply:
Yeah, sounds good -- done!